listener/src/event_handler/flux_transition_handler.rs

```diff
@@ -179,6 +185,79 @@ pub mod test {
         assert!(result.is_err());
     }
 
+    #[test]
+    fn test_applies_withdraws_existing_state() {
+        let schema = util::test::fake_schema();
+        let store = Arc::new(MockStore::new(vec![(schema.id.clone(), schema)]));
+        let handler = FluxTransitionHandler::new(store.clone());
+
+        // Add previous account state and pending deposits into Store
+        let existing_state = AccountState::new(
+            H256::zero(),
+            U256::zero(),
+            vec![10, 20, 0, 0],
+            1
+        );
+        let entity = existing_state.into();
+        store.apply_entity_operations(vec![EntityOperation::Set {
+            key: util::entity_key("AccountState", &entity),
+            data: entity
+        }], None).unwrap();
+        let first_withdraw = PendingFlux {
+            slot_index: 0,
+            slot: U256::zero(),
+            account_id: 0,
+            token_id: 0,
+            amount: 10,
+        };
+        let mut entity: Entity = first_withdraw.into();
+        entity.set("id", "1");
+        store.apply_entity_operations(vec![EntityOperation::Set {
+            key: util::entity_key("Withdraw", &entity),
+            data: entity
+        }], None).unwrap();
+
+        let second_withdraw = PendingFlux {
```

Comment:
Maybe include one withdraw that's larger than the balance and thus should get ignored?

Reply:
Yes this is certainly something that should be tested for!

Reply:
Done!